src/compiler/checker.ts

```diff
@@ -4504,18 +4527,30 @@ namespace ts {
                         parameters.push(paramSymbol);
                     }
 
-                    if (param.type && param.type.kind === SyntaxKind.StringLiteralType) {
-                        hasStringLiterals = true;
-                    }
-
-                    if (param.initializer || param.questionToken || param.dotDotDotToken || isJSDocOptionalParameter(param)) {
-                        if (minArgumentCount < 0) {
-                            minArgumentCount = i - (hasThisParameter ? 1 : 0);
+                    if (isJSDocParameterTag(param)) {
+                        if (isJSDocOptionalParameterTag(param)) {

```

Comment:
this is *really* almost the same as the code in the `!isJSDocParameterTag` branch. I think you should get rid of the true branch and find a way to make the already-existing `isJSDocOptionalParameterTag` (currently, line 4545) check work.